### PR TITLE
Pods 1156: Prevent users from submitting incomplete data

### PR DESCRIPTION
**Pull request by darrylmurphy**

### CODE REVIEW

File: app/controllers/register/company/AddCompanyDirectorsController.scala [hunk 1]
```diff
@@ -78,4 +84,26 @@ class AddCompanyDirectorsController @Inject() (
...
+  private def directorsWithFlag(
+                                 withFlag: Seq[(DirectorDetails, Boolean)],
+                                 directors: Seq[DirectorDetails],
+                                 flags: Seq[Boolean]
+                               ): Seq[(DirectorDetails, Boolean)] = {
+
+    if(directors.isEmpty){
+      withFlag
+    } else if (flags.isEmpty) {
+      val addWithFlag: (DirectorDetails, Boolean) = (directors.head, false)
```
Comment (versatil56) [inline app/controllers/register/company/AddCompanyDirectorsController.scala]: We don't really need to load this recurively, we could just assess  whether if we have `flags` beforehand and then just map over directors to return Seq[(Directors,boolean)]

File: app/controllers/register/company/AddCompanyDirectorsController.scala [hunk 2]
```diff
@@ -78,4 +84,26 @@ class AddCompanyDirectorsController @Inject() (
...
+      directorsWithFlag(withFlag :+ addWithFlag, directors.tail, Seq.empty)
+    } else {
+      val addWithFlag: (DirectorDetails, Boolean) = (directors.head, flags.head)
```
Comment (versatil56) [inline app/controllers/register/company/AddCompanyDirectorsController.scala]: Could we achieve the same with foldLeft? If not, we could set this method as @tailrec then the compiler attempts to optimize it.

File: app/controllers/register/company/AddCompanyDirectorsController.scala [hunk 3]
```diff
@@ -78,4 +84,26 @@ class AddCompanyDirectorsController @Inject() (
...
+      directorsWithFlag(withFlag :+ addWithFlag, directors.tail, flags.tail)
+    }
+
+  }
+
+  private def disableSubmission(directorsWithFlag: Seq[(DirectorDetails, Boolean)]): Boolean = directorsWithFlag.foldLeft(true){ (x, y) =>
+    !y._2
```
Comment (versatil56) [inline app/controllers/register/company/AddCompanyDirectorsController.scala]: we don't really need to declare `x`, maybe it'd be good if we renamed `y` to `isDeleted` or something like that -- just for readability.

File: app/views/components/person_list.scala.html [hunk 4]
```diff
@@ -14,24 +14,44 @@
...
             <div class="cya-question">
                 <h2 id="@person.id" class="bold std-text">@person.name</h2>
             </div>
 
-            <div class="cya-change">
+            @if(appConfig.completeFlagEnabled){
+
+                @if(flag){
+
+                <div class="cya-status status">
```
Comment (versatil56) [inline app/views/components/person_list.scala.html]: `<div class="cya-status status"> </div>` can be set in the outter scope of `if(flag)`

File: test/controllers/actions/AuthActionSpec.scala [hunk 5]
```diff
@@ -213,7 +213,6 @@ object AuthActionSpec {
 
   private def appConfig(isSchemeOverviewEnabled: Boolean = false) = {
     val application = new GuiceApplicationBuilder()
```
Comment (versatil56) [inline test/controllers/actions/AuthActionSpec.scala]: I removed this configure statement as it is not needed anymore. Please pull latest on your code.

File: app/controllers/register/company/AddCompanyDirectorsController.scala (hunk 3 above)
Comment (darrylmurphy) [reply app/controllers/register/company/AddCompanyDirectorsController.scala]: Done

File: app/controllers/register/company/AddCompanyDirectorsController.scala (hunk 2 above)
Comment (darrylmurphy) [reply app/controllers/register/company/AddCompanyDirectorsController.scala]: @tailrec added
The reason a map or foldLeft or indeed zip, zipAll have not been used as we iterate and combine two separate collections. It should continue to iterate over directors adding a false flag as default.

File: app/views/components/person_list.scala.html (hunk 4 above)
Comment (darrylmurphy) [reply app/views/components/person_list.scala.html]: Better to hide an empty column when toggle is off and only show the populated column if the toggle is switched on? Stylesheets may make a visible difference

**Pull request closed without merging**